src/main/java/org/apache/sling/resourceresolver/impl/ResourceResolverMetrics.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.resourceresolver.impl;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.function.Supplier;
+
+import org.apache.sling.commons.metrics.Counter;
+import org.apache.sling.commons.metrics.Gauge;
+import org.apache.sling.commons.metrics.MetricsService;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(service=ResourceResolverMetrics.class, immediate=true)
+public class ResourceResolverMetrics {
+    
+    protected static final String METRICS_PREFIX = "org.apache.sling.resourceresolver";
+    
+    @Reference
+    MetricsService metricsService;
+    
+    private static final Supplier<Long> ZERO_SUPPLIER = () -> 0L;
+    
+    // number of vanity paths
+    private ServiceRegistration<Gauge<Long>> numberOfVanityPathsGauge;
+    private Supplier<Long> numberOfVanityPathsSupplier = ZERO_SUPPLIER;
+    
+    // number of aliases
+    private ServiceRegistration<Gauge<Long>> numberOfAliasesGauge;
+    private Supplier<Long> numberOfAliasesSupplier = ZERO_SUPPLIER;
+    
+    private Counter unclosedResourceResolvers;
+    
+    
+    @Activate
+    protected void activate(ComponentContext context) {
+        BundleContext bundleContext = context.getBundleContext();
+        numberOfVanityPathsGauge = registerGauge(bundleContext, METRICS_PREFIX + ".numberOfVanityPaths", () -> numberOfVanityPathsSupplier );
```

Comment:
Can we avoid dealing with framework api?

Reply:
There is no equivalent of ```metricsService.counter()``` for Gauge :-/ , but only https://issues.apache.org/jira/browse/SLING-5966

Reply:
Right, but the two gauge services could be DS components with a reference to the ResourceResolverMetrics

Reply:
From my point of view it would get more complex and require more code, and I don't see any benefit doing so.

Reply:
Mixin DS and framework api is an anti pattern; while I don't agree that using DS is substantially more code or more complex, I guess we can leave it like this for now

Reply:
I raised https://issues.apache.org/jira/browse/SLING-10967 to move this logic into Sling Commons Metrics. I will keep this logic here until the change in the Sling Commons Metric is implemented.